crates/aptos-rest-client/src/lib.rs

```diff
@@ -73,30 +73,11 @@ pub struct Client {
 
 impl Client {
     pub fn builder(aptos_base_url: AptosBaseUrl) -> ClientBuilder {
-        ClientBuilder::new(aptos_base_url).unwrap()
+        ClientBuilder::new(aptos_base_url)
     }
 
     pub fn new(base_url: Url) -> Self {
-        let mut header_map = HeaderMap::new();
-        header_map.insert(
-            HeaderName::from_static(X_APTOS_CLIENT),
-            HeaderValue::from_static(X_APTOS_SDK_HEADER_VALUE),
-        );
-
-        let inner = ReqwestClient::builder()
-            .timeout(Duration::from_secs(10)) // Default to 10 seconds
-            .default_headers(header_map)
-            .cookie_store(true)
-            .build()
-            .unwrap();
-
-        let version_path_base = get_version_path_with_base(base_url.clone());
-
-        Self {
-            inner,
-            base_url,
-            version_path_base,
-        }
+        Self::builder(AptosBaseUrl::Custom(base_url)).build()
```

Comment:
Cleaaaaaaaannnnnn!!!!!!!!

Saying that, I think we still need to use `get_version_path_with_base` otherwise the URL might be missing `/v1`. But rather than putting it here let's do it in the `ClientBuilder`.

Reply:
Oop Jin pointed out to me we are indeed doing this in the ClientBuilder!